File: assignment1.java
//shell sort in java

import java.util.Scanner;
public class ShellSort {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter the size of array:");
        int size = scanner.nextInt();

        int[] arr = new int[size];

        System.out.println("Enter the elements to sort:");
        for(int i=0; i<size; i++){
            arr[i] = scanner.nextInt();
        }
        ShellSort(arr);
        System.out.println("Array after sort:");
        for(int i=0; i<size; i++){
            System.out.print(arr[i] + " ");
        }
    scanner.close();
}

public static void shellSort(int[] arr){
    int n = arr.length;
    for(int gap = n/2 ; gap > 0 ; gap/=2){
        for(int i=gap ; i<n ; i++){
            int temp = arr[i];
            int j;
            for(j=i;j>=gap && arr[j-gap]>temp; j-=gap){
                arr[j] = arr[j-gap];
            }
            arr[j] = temp;
        }
    }
}
}

File: assignment1.cpp
//Shell sort in C++

#include <iostream>
using namespace std;
 void shellSort(int arr[] , int n){
    for(int gap=n/2; gap>0; gap/=2){
        for(int i=gap; i<n; i++){
            int temp = arr[i];
            int j;
            for(j=i; j>=gap && arr[j-gap]>temp; j-=gap){
                arr[j] = arr[j-gap];
            }
            arr[j] = temp;
        }
    }
 }

 int main(){
    int n;
    cout<<"\nEnter the size of the array:";
    cin>>n;
    int arr[n];

    cout<<"\nEnter the elements to sort:";
    for(int i=0; i<n; i++){
        cin>>arr[i];
    }

    shellSort(arr,n);

    cout<<"\nElements after sorting:"<<endl;
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
 }
